### PR TITLE
Added news controller and `/personalized` route for fetching personalized news

### PR DESCRIPTION
**Pull request by bojana-petroska**
This PR adds a new `/personalized` route to fetch news based on user preferences. The route uses `auth` middleware and accepts categories (required) and limit (optional, default: 10) as query parameters.

**Key changes:**

- Added `/personalized` route in the router.
- Implemented `getPersonalizedNews` to validate input, fetch news from NewsAPI, and format the response.
Included error handling for missing parameters `400` and server errors `500`.

### CODE REVIEW

File: src/modules/News/controller.ts [hunk 1]
```diff
@@ -13,4 +13,36 @@ const getLatestNews = async (req: Request, res: Response) => {
   }
 };
 
-export default { getLatestNews };
+const getPersonalizedNews = async (req: Request, res: Response) => {
+  const { categories, limit = 10 } = req.query;
```
Comment (ahzia) [inline src/modules/News/controller.ts:17]: assign the default value in the validation schema

File: src/modules/News/controller.ts [hunk 2]
```diff
@@ -13,4 +13,36 @@ const getLatestNews = async (req: Request, res: Response) => {
...
+
+  if (!categories) {
+    res
+      .status(400)
+      .send({ message: 'Missing required query parameter: categories' });
+    return;
+  }
```
Comment (ahzia) [inline src/modules/News/controller.ts:24]: this will be done during the validation middleware

File: src/modules/News/controller.ts [hunk 3]
```diff
@@ -13,4 +13,36 @@ const getLatestNews = async (req: Request, res: Response) => {
...
+
+  try {
+    const news = await fetchFromNewsAPI('/everything', {
+      q: categories,
+      limit,
+    });
+
+    const formattedResponse = news.articles.map((article: any) => ({
```
Comment (ahzia) [inline src/modules/News/controller.ts:32]: Suggestion: add a type for `article `

File: src/modules/News/routes.ts [hunk 4]
```diff
@@ -3,11 +3,13 @@ import newsValidationSchemas from './validation.js';
...
 import newsController from './controller.js';
 const router = Router();
+import auth from '../../middleware/auth.js';
 
 router.get(
   '/latest',
   validateRequest(newsValidationSchemas.latestNews),
   newsController.getLatestNews,
 );
+router.get('/personalized', auth, newsController.getPersonalizedNews);
```
Comment (ahzia) [inline src/modules/News/routes.ts:13]: Use validation middleware and validate queries.

**Review by ahzia — CHANGES_REQUESTED** [review]
Good Job @bojana-petroska 
Take a look at my inline comments, and add validation.

**Comment by bojana-petroska** [comment]
This pull request was closed and the new one is: [new pull request](https://github.com/StartSteps-Digital-Education-GmbH/News-App/pull/14)

**Pull request closed without merging**